website/static/js/tests/registrationUtils.test.js

```diff
@@ -0,0 +1,448 @@
+/*global describe, it, expect, example, before, after, beforeEach, afterEach, mocha, sinon*/
+'use strict';
+var assert = require('chai').assert;
+var $ = require('jquery');
+var utils = require('tests/utils');
+var faker = require('faker');
+
+var bootbox = require('bootbox');
+
+var $osf = require('js/osfHelpers');
+
+window.contextVars.currentUser = {
+    fullname: faker.name.findName(),
+    id: 1
+};
+var registrationUtils = require('js/registrationUtils');
+
+var utilities = registrationUtils.utilities;
+var Comment = registrationUtils.Comment; // jshint ignore:line
+var Question = registrationUtils.Question;
+var MetaSchema = registrationUtils.MetaSchema;
+var Draft = registrationUtils.Draft;
+var RegistrationEditor = registrationUtils.RegistrationEditor;
+var RegistrationManager = registrationUtils.RegistrationManager;
+
+var mkMetaSchema = function() {
+    var questions = {};
+    var qid;
+    [1, 1, 1].map(function() {
+        qid = faker.internet.ip();
+        questions[qid] = {
+            type: 'string',
+            format: 'text'
+        };
+    });
+
+    var params = {
+        schema_name: 'My Schema',
+        schema_version: 1,
+        title: 'A schema',
+        schema: {
+            title: 'A schema',
+            version: 1,
+            description: 'A very interesting schema',
+            fulfills: [],
+            pages: [1, 1, 1].map(function() {
+                return {
+                    id: faker.internet.ip(),
+                    title: 'Page',
+                    questions: questions
+                };
+            })
+        },
+        id: 'asdfg'
+    };
+
+    var ms = new MetaSchema(params);
+    return [qid, params, ms];
+};
+
+describe('Question', () => {
+    var id, question, q;
+    beforeEach(() => {
+        id = faker.internet.ip();
+        question = {
+            title: faker.internet.domainWord(),
+            nav: faker.internet.domainWord(),
+            type: 'string',
+            format: 'text',
+            description: faker.lorem.sentence(),
+            help: faker.lorem.sentence(),
+            required: true,
+            options: [1, 1, 1].map(faker.internet.domainWord)
```

Comment:
Please try to avoid using `.map` as a shorthand for looping; these sorts of shortcuts save a few letters of typing at the expense of readability.*

* As decided by a randomly chosen sample of OSF developers.